### DISCUSSION TITLE
Next-Auth with Django, how can we do it? Email Based, and OAuth Both!

### DISCUSSION THREAD

**Discussion by Bishwas-py** [vercel/next.js#36494] [Help]
I have be struggling a lot with connecting NextAuth Oauth with Django Backend, also I am really confused how we can de Email/Password based authentication with with NextAuth (Next.Js) and Django Rest Framework backed... Has anyone have ever done it?

**Comment by icyJoseph** [comment]
Hi,

Not a very helpful answer here, but you might as well try your luck over here: https://github.com/nextauthjs/next-auth/discussions/1350, perhaps even open a discussion question as well.

**Comment by mrob11** [comment]
I wrote an article about doing this if you're still interested. It's a two-parter. Here's part one.

https://heymike.dev/articles/using-django-as-openid-connect-provider
